services/provider/server/server.go

```diff
@@ -206,3 +212,121 @@ func Start(port int, providerServer *OCSProviderServer, opts []grpc.ServerOption
+	}
+
+	cmData := strings.Split(configmap.Data["data"], ",")
+	data := map[string]string{
+		"data":     cmData[0],
+		"maxMonId": "0",
+		"mapping":  "{}",
+	}
+	name := "rook-ceph-mon-endpoints"
+	kind := "ConfigMap"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Secret storing cluster mon.admin key, fsid and name
+	scMon, err := s.consumerAPI.GetSecret("rook-ceph-mon", s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-mon` secret. Error %v", err)
+	}
+	data = map[string]string{
+		"fsid":         string(scMon.Data["fsid"]),
+		"mon-secret":   "mon-secret",
+		"admin-secret": "admin-secret",
+	}
+	name = "rook-ceph-mon"
+	kind = "Secret"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Ceph user secret
+	storageConsumer, err := s.consumerAPI.GetStorageConsumer(consumerResource.Name, s.namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get storageConsumer resource for the given consumer: %v", err)
+	}
+	cephUser := storageConsumer.Status.CephObjects.CephUser
+	scCephUser, err := s.consumerAPI.GetSecret(fmt.Sprintf("rook-ceph-client-%s", cephUser), s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-client %s` secret. Error %v", "", err)
+	}
+
+	data = map[string]string{
+		"userID":  "client." + cephUser,
+		"userKey": string(scCephUser.Data[cephUser]),
+	}
+	name = "rook-ceph-operator-creds"
+	kind = "Secret"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Secret storing CSI RBD node userID and key
+	data = map[string]string{
+		"userID":  "csi-rbd-node",
+		"userKey": string(scCephUser.Data[cephUser]),
+	}
+	name = "rook-csi-rbd-node"
+	kind = "Secret"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Secret storing CSI RBD provisioner userID and key
+	data = map[string]string{
+		"userID":  "csi-rbd-provisioner",
+		"userKey": string(scCephUser.Data[cephUser]),
+	}
+	name = "rook-csi-rbd-provisioner"
+	kind = "Secret"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Service for monitoring endpoints
+	scMonitoring, err := s.consumerAPI.GetService("rook-ceph-mgr", s.namespace)
+	if err != nil {
+		return []externalSC.ExternalResource{}, fmt.Errorf("failed to get `rook-ceph-mgr` service. Error %v", err)
+	}
+
+	data = map[string]string{
+		"MonitoringEndpoint": scMonitoring.Spec.ClusterIP,
+		"MonitoringPort":     fmt.Sprintf("%v", scMonitoring.Spec.Ports[0].Port),
+	}
+	name = "monitoring-endpoint"
+	kind = "CephCluster"
+
+	j = append(j, externalSC.ExternalResource{Name: name, Kind: kind, Data: data})
+
+	// Service having mgr ip and port for ceph dashboard
+	scDashboard, err := s.consumerAPI.GetService("rook-ceph-mgr-dashboard", s.namespace)
```

Comment:
Why do we need this? What is using the ceph dashboard?

Reply:
so, if we are not using than I'll remove.